drivers/flash/soc_flash_nrf.c

```diff
@@ -153,17 +175,28 @@ static int flash_nrf_erase(struct device *dev, off_t addr, size_t size)
 	u32_t n_pages = size / pg_size;
 	int ret;
 
-	/* Erase can only be done per page */
-	if (((addr % pg_size) != 0) || ((size % pg_size) != 0)) {
-		return -EINVAL;
-	}
+#ifndef CONFIG_SOC_FLASH_NRF_UICR
+	if (is_addr_valid(addr, size)) {
+#else
 
-	if (!is_addr_valid(addr, size)) {
-		return -EINVAL;
-	}
+	if (is_regular_addr_valid(addr, size)) {
+#endif
+		/* Erase can only be done per page */
+		if (((addr % pg_size) != 0) || ((size % pg_size) != 0)) {
+			return -EINVAL;
+		}
 
-	if (!n_pages) {
-		return 0;
+		if (!n_pages) {
+			return 0;
+		}
+	} else {
+#ifndef CONFIG_SOC_FLASH_NRF_UICR
```

Comment:
switch sense of test as above

Reply:
changes all ne #ifdef-ry to possitive wise